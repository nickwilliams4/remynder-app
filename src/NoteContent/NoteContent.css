.NoteContentPage {
  margin-top: 40px;
}

.Note_title {
  font-size: 30px;
  text-decoration: underline;
}

.NoteContent {
  margin-top: 50px;
  font-size: x-large;
}

.Note__dates-created {
  border-top: groove;
  margin-top: 75px;
}

.back-button {
    height: 40px;
    width: 120px;
    border-radius: 20px;
    font-size: 16px;
    background-color: black;
    color: white;
  }
  
.back-button-icon {
  margin-right: 10px;
}

.back-button:hover {
  color: black;
  background-color: white;
}