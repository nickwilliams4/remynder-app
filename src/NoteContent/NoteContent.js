import React from 'react'
import ApiContext from '../ApiContext'
import { findNote } from '../HelperFunctions'
import './NoteContent.css'
import moment from 'moment'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default class NoteContent extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { }
    },
    match: {
      params: {}
    }
  }
  static contextType = ApiContext;

  render() {
    const { notes } = this.context
    const { note_id } = this.props.match.params
    const note = findNote(notes, +note_id) || {}
    return (
      <div className='NoteContentPage'>
        <div className='Note_title'>{note.title}</div>
        <h4 className='NoteContent'>{note.content}</h4>
        <div className='Note__dates'>
          <div className='Note__dates-created'>
            Created:
          {' '}
            <span className='Date'>
              {moment(note.created).format('MMM Do YYYY')}
            </span>
          </div>
        </div>
        <p>Remynder: Every {note.remynder} hours</p>
        <Link to='/NotePage'>
          <button className='back-button'>
            <FontAwesomeIcon icon='angle-double-left' className="back-button-icon"/>
            Back
          </button>
        </Link>
      </div>
    )
  }
}