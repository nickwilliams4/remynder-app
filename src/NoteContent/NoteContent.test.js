import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteContent from './NoteContent'

describe(`NoteContent component`, () => {
  const props = {
    note_id: 'a',
    name: 'test-class-name',
  }

  it('renders the NoteContent', () => {
    const wrapper = shallow(<NoteContent {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})