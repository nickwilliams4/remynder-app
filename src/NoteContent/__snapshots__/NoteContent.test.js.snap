// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteContent component renders the NoteContent 1`] = `
<div
  className="NoteContentPage"
>
  <div
    className="Note_title"
  />
  <h4
    className="NoteContent"
  />
  <div
    className="Note__dates"
  >
    <div
      className="Note__dates-created"
    >
      Created:
       
      <span
        className="Date"
      >
        Apr 20th 2020
      </span>
    </div>
  </div>
  <p>
    Remynder: Every 
     hours
  </p>
  <Link
    to="/NotePage"
  >
    <button
      className="back-button"
    >
      <FontAwesomeIcon
        border={false}
        className="back-button-icon"
        fixedWidth={false}
        flip={null}
        icon="angle-double-left"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      Back
    </button>
  </Link>
</div>
`;
